plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/util/SynthesizedColumnHandler.java

```diff
@@ -0,0 +1,191 @@
+import static io.trino.plugin.hive.HiveColumnHandle.FILE_SIZE_COLUMN_NAME;
+import static io.trino.plugin.hive.HiveColumnHandle.PARTITION_COLUMN_NAME;
+import static io.trino.plugin.hive.HiveColumnHandle.PATH_COLUMN_NAME;
+import static io.trino.spi.type.DateTimeEncoding.packDateTimeWithZone;
+import static io.trino.spi.type.TimeZoneKey.UTC_KEY;
+
+/**
+ * Handles synthesized (virtual) columns in Hudi tables, such as partition columns and metadata (not hudi metadata)
+ * columns.
+ */
+public class SynthesizedColumnHandler
+{
+    private final Map<String, SynthesizedColumnStrategy> strategies;
+    private final SplitMetadata splitMetadata;
+
+    public static SynthesizedColumnHandler create(HudiSplit hudiSplit)
+    {
+        return new SynthesizedColumnHandler(hudiSplit);
+    }
+
+    /**
+     * Constructs a SynthesizedColumnHandler with the given partition keys.
+     */
+    public SynthesizedColumnHandler(HudiSplit hudiSplit)
+    {
+        this.splitMetadata = SplitMetadata.of(hudiSplit);
+        ImmutableMap.Builder<String, SynthesizedColumnStrategy> builder = ImmutableMap.builder();
+        initSynthesizedColStrategies(builder);
+        initPartitionKeyStrategies(builder, hudiSplit);
+        strategies = builder.buildOrThrow();
+    }
+
+    /**
+     * Initializes strategies for synthesized columns.
+     */
+    private void initSynthesizedColStrategies(ImmutableMap.Builder<String, SynthesizedColumnStrategy> builder)
+    {
+        builder.put(PARTITION_COLUMN_NAME, blockBuilder ->
+                VarcharType.VARCHAR.writeSlice(blockBuilder,
+                        utf8Slice(toPartitionName(splitMetadata.getPartitionKeyVals()))));
+
+        builder.put(PATH_COLUMN_NAME, blockBuilder ->
+                VarcharType.VARCHAR.writeSlice(blockBuilder, utf8Slice(splitMetadata.getFilePath())));
+
+        builder.put(FILE_SIZE_COLUMN_NAME, blockBuilder ->
+                BigintType.BIGINT.writeLong(blockBuilder, splitMetadata.getFileSize()));
+
+        builder.put(FILE_MODIFIED_TIME_COLUMN_NAME, blockBuilder -> {
+            long packedTimestamp = packDateTimeWithZone(
+                    splitMetadata.getFileModificationTime(), UTC_KEY);
+            TimestampWithTimeZoneType.TIMESTAMP_TZ_MILLIS.writeLong(blockBuilder, packedTimestamp);
+        });
+    }
+
+    /**
+     * Initializes strategies for partition columns.
+     */
+    private void initPartitionKeyStrategies(ImmutableMap.Builder<String, SynthesizedColumnStrategy> builder,
+            HudiSplit hudiSplit)
+    {
+        for (HivePartitionKey partitionKey : hudiSplit.partitionKeys()) {
+            builder.put(partitionKey.name(), (blockBuilder) ->
+                    VarcharType.VARCHAR.writeSlice(blockBuilder, utf8Slice(partitionKey.value())));
+        }
+    }
+
+    /**
+     * Checks if a column is a synthesized column.
+     *
+     * @param columnName The column name.
+     * @return True if the column is synthesized, false otherwise.
+     */
```

Comment:
This is obvious from the method name. We can remove this javadoc. Same for others.